src/time/time_format/formatter.cr

```diff
@@ -176,6 +176,101 @@ struct TimeFormat
       {negative, hours, minutes}
     end
 
+    # Compute week of the year, monday is the first day of the week
+    def week_of_year_monday_1_7
+      io << Formatter.iso8601_week_of_year(time, 1)
+    end
+
+    # Compute week of the year, sunday is the first day of the week
+    def week_of_year_sunday_0_6
+      io << Formatter.iso8601_week_of_year(time, 0)
+    end
+
+    # Internal helper wethod: Figure how many weeks into the year
+    def self.week_of_year(time, firstweekday)
+      day_of_week = TimeFormat.new("%w").format(time).to_i
+      week_num = 0
+
+      if firstweekday == 1
+        if day_of_week == 0 # sunday
+          day_of_week = 6
+        else
+          day_of_week -= 1
+        end
+      end
+
+      day_of_year = TimeFormat.new("%j").format(time).to_i
+      week_num = ((day_of_year + 7 - day_of_week) / 7)
+      week_num = 0 if week_num < 0
+
+      week_num
+    end
+
+    # Compute week number according to ISO 8601
+    def self.iso8601_week_of_year(time, firstweekday)
+      #	If the week (Monday to Sunday) containing January 1
+      #	has four or more days in the new year, then it is week 1;
+      #	otherwise it is the highest numbered week of the previous
+      #	year (52 or 53), and the next week is week 1.
+
+      weeknum = 0
+
+      # Get week number
+      weeknum = Formatter.week_of_year(time, firstweekday)
+
+      # What day of the week does January 1 fall on?
+      day_of_week_format = firstweekday == 0 ? TimeFormat.new("%w") : TimeFormat.new("%u")
+
+      jan1day = day_of_week_format.format(time).to_i - (TimeFormat.new("%j").format(time).to_i % 7)
```

Comment:
Already changed it in the formatter ... the firstweekday thingy is annoying as hell ...